### PR TITLE
Prefactorings for module augmentation

### PR DESCRIPTION
**Pull request by nvie**
Prefactorings for the upcoming module augmentation adaption work, which will be noisy.

All changes in this PR **don't change any behavior**, and should be **good to be merged** into `main` right now.

### CODE REVIEW

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 1]
```diff
@@ -46,16 +46,100 @@ export const ContextBundle = createContext<LiveblocksContextBundle<
...
+    raise("LiveblocksProvider is missing from the React tree.")
+  );
 }
 
 export const POLLING_INTERVAL = 60 * 1000; // 1 minute
 export const INBOX_NOTIFICATIONS_QUERY = "INBOX_NOTIFICATIONS";
 
+// --- Selector helpers ------------------------------------------------- {{{
+
+function selectorFor_useInboxNotifications(
```
Comment (nvie) [inline packages/liveblocks-react/src/liveblocks.tsx:60]: I've renamed these helpers to not start with `use*`, as they aren't hooks.

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 2]
```diff
@@ -193,55 +255,46 @@ export function createLiveblocksContext<
...
+    }, [autoFetch]);
   }
 
   function useInboxNotifications(): InboxNotificationsState {
-    useEffect(() => {
-      void fetchInboxNotifications();
-      incrementInboxNotificationsSubscribers();
-
-      return () => decrementInboxNotificationsSubscribers();
-    }, []);
```
Comment (nvie) [inline packages/liveblocks-react/src/liveblocks.tsx:226]: This block is now abstracted as a custom hook `useSubscribeToInboxNotificationsEffect()`. There is no direct access to the inc/dec functions anymore.

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 3]
```diff
@@ -460,27 +443,23 @@ export function createLiveblocksContext<
   ): ThreadData<TThreadMetadata> {
     const selector = useCallback(
       (state: CacheState<TThreadMetadata>) => {
-        const inboxNotification = state.inboxNotifications[inboxNotificationId];
-
-        if (inboxNotification === undefined) {
-          throw new Error(
+        const inboxNotification =
+          state.inboxNotifications[inboxNotificationId] ??
+          raise(
```
Comment (nvie) [inline packages/liveblocks-react/src/liveblocks.tsx:448]: Just using `... ?? raise(...)` for brevity here.

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 4]
```diff
@@ -495,8 +474,7 @@ export function createLiveblocksContext<
     );
   }
 
-  const currentUserIdStore = client[kInternal]
-    .currentUserIdStore as unknown as Store<string | null>;
```
Comment (nvie) [inline packages/liveblocks-react/src/liveblocks.tsx:499]: This type cast seemed to not be needed.

File: packages/liveblocks-react/tsconfig.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
     "target": "es2020",
     "module": "es2020",
     "lib": ["dom"],
+    "skipLibCheck": true,
```
Comment (nvie) [inline packages/liveblocks-react/tsconfig.json:9]: Was seeing some type errors inside `node_modules`, so I've disabled them here.

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 6]
```diff
@@ -46,16 +46,100 @@ export const ContextBundle = createContext<LiveblocksContextBundle<
...
+  return (
+    useContext(ContextBundle) ??
+    raise("LiveblocksProvider is missing from the React tree.")
+  );
 }
 
 export const POLLING_INTERVAL = 60 * 1000; // 1 minute
 export const INBOX_NOTIFICATIONS_QUERY = "INBOX_NOTIFICATIONS";
 
+// --- Selector helpers ------------------------------------------------- {{{
```
Comment (nvie) [inline packages/liveblocks-react/src/liveblocks.tsx:58]: All of these helpers haven't been changed. They're just moved out of the closure and into the top-level, which will make it easier to reuse them in the top-level hooks soon.

File: packages/liveblocks-react/tsconfig.json (hunk 5 above)
Comment (GuillaumeSalles) [reply packages/liveblocks-react/tsconfig.json:9]: Do you remember which ones?

**Review by GuillaumeSalles — APPROVED** [review]
🚀

**Review by marcbouchenoire — APPROVED** [review]
🔜

**Comment by nvie** [comment]
> _"Do you remember which ones?"_

Yeah, they were coming from `jest` and from `date-fns`, and I didn't feel like investigating now. [Full list of errors.](https://gist.github.com/nvie/4a53eaa2cd3981d118583155f6b8b3bd)

**Pull request merged**